# T-100 Data Acquisition

This notebook documents and verifies the acquisition and initial loading of BTS T-100 International Segment data for the Airline Passenger Demand Forecasting & Capacity Planning project.

In [27]:
import sys

print(sys.executable)
print(sys.version)

c:\Users\PC\airline-demand-capacity-planning\.venv\Scripts\python.exe
3.12.2 (tags/v3.12.2:6abddd9, Feb  6 2024, 21:26:36) [MSC v.1937 64 bit (AMD64)]


In [28]:
import pandas as pd

print("pandas:", pd.__version__)

pandas: 3.0.5


In [29]:
raw_file_path = "../data/raw/T_T100I_SEGMENT_ALL_CARRIER_20260907_133912.zip"

In [30]:
from pathlib import Path
from zipfile import ZipFile

raw_path = Path(raw_file_path)

print("Archive exists:", raw_path.exists())
print("Archive size (bytes):", raw_path.stat().st_size)

with ZipFile(raw_path, "r") as archive:
    print("Files inside archive:")
    for file_info in archive.infolist():
        print(
            file_info.filename,
            "- uncompressed bytes:", file_info.file_size,
            "- compressed bytes:", file_info.compress_size
        )

Archive exists: True
Archive size (bytes): 349414
Files inside archive:
T_T100I_SEGMENT_ALL_CARRIER.csv - uncompressed bytes: 2550165 - compressed bytes: 349254


In [31]:
df = pd.read_csv(raw_file_path)
print('type: ' , type(df))
print('shape: ' , df.shape)

type:  <class 'pandas.DataFrame'>
shape:  (9018, 43)


In [32]:
for i,column in enumerate(df.columns, start=1):
    print(i, column)

1 DEPARTURES_SCHEDULED
2 DEPARTURES_PERFORMED
3 PAYLOAD
4 SEATS
5 PASSENGERS
6 FREIGHT
7 MAIL
8 DISTANCE
9 RAMP_TO_RAMP
10 AIR_TIME
11 UNIQUE_CARRIER
12 AIRLINE_ID
13 UNIQUE_CARRIER_NAME
14 UNIQUE_CARRIER_ENTITY
15 REGION
16 CARRIER
17 CARRIER_NAME
18 CARRIER_GROUP
19 CARRIER_GROUP_NEW
20 ORIGIN_AIRPORT_ID
21 ORIGIN_AIRPORT_SEQ_ID
22 ORIGIN_CITY_MARKET_ID
23 ORIGIN
24 ORIGIN_CITY_NAME
25 ORIGIN_COUNTRY
26 ORIGIN_COUNTRY_NAME
27 ORIGIN_WAC
28 DEST_AIRPORT_ID
29 DEST_AIRPORT_SEQ_ID
30 DEST_CITY_MARKET_ID
31 DEST
32 DEST_CITY_NAME
33 DEST_COUNTRY
34 DEST_COUNTRY_NAME
35 DEST_WAC
36 AIRCRAFT_GROUP
37 AIRCRAFT_TYPE
38 AIRCRAFT_CONFIG
39 YEAR
40 QUARTER
41 MONTH
42 DISTANCE_GROUP
43 CLASS


In [33]:
key_columns = ['UNIQUE_CARRIER', 'YEAR', 'MONTH', 'ORIGIN', 'DEST', 'PASSENGERS', 'SEATS', 'DEPARTURES_SCHEDULED', 'DEPARTURES_PERFORMED', 'DISTANCE']
print(df[key_columns].dtypes)

UNIQUE_CARRIER              str
YEAR                      int64
MONTH                     int64
ORIGIN                      str
DEST                        str
PASSENGERS              float64
SEATS                   float64
DEPARTURES_SCHEDULED    float64
DEPARTURES_PERFORMED    float64
DISTANCE                float64
dtype: object


In [34]:
df[key_columns].head()

,UNIQUE_CARRIER,YEAR,MONTH,ORIGIN,DEST,PASSENGERS,SEATS,DEPARTURES_SCHEDULED,DEPARTURES_PERFORMED,DISTANCE
0,DL,2024,1,YUL,HPN,0.0,0.0,0.0,0.0,304.0
1,36Q,2024,1,SCF,NAS,0.0,0.0,0.0,1.0,2150.0
2,G3,2024,1,PAE,PUJ,0.0,0.0,0.0,1.0,3621.0
3,G3,2024,1,MCO,CNF,0.0,0.0,0.0,1.0,4141.0
4,G3,2024,1,GRU,MCO,0.0,0.0,0.0,1.0,4255.0


In [35]:
print('YEAR: ', df['YEAR'].unique())
print('MONTH: ', df['MONTH'].unique())

YEAR:  [2024]
MONTH:  [1]


In [36]:
print(set(key_columns) - set(df.columns.tolist()))

set()


## Initial Acquisition Verification

- Source: U.S. DOT Bureau of Transportation Statistics (BTS)
- Table: T-100 International Segment (All Carriers)
- Carrier scope: ALL CARRIERS
- Requested period: January 2024
- Raw archive: T_T100I_SEGMENT_ALL_CARRIER_20260907_133912.zip
- File inside archive: T_T100I_SEGMENT_ALL_CARRIER.csv
- DataFrame shape: (9018, 43)
- Number of columns: 43
- Required project fields present: Yes

## Source Field Mapping
|BTS CSV Column| BTS Meaning | Project Concept|
|--------------|--------------|----------------|
|UNIQUE_CARRIER| Unique Carrier Code | Carrier Code|
|YEAR| Year |Year of observation|
|MONTH| Month Of Flight|Month of observation|
|ORIGIN| Origin Airport|Directional route origin|
|DEST| Destination Airport| Directional route destination|
|PASSENGERS| non-stop segment passengers transported | Observed passenger volume|
|SEATS| Available Seats| Historical available seat capacity
| DEPARTURES_SCHEDULED | Departures Scheduled | Scheduled flight departures |
| DEPARTURES_PERFORMED | Departures Performed | Actual performed departures |
| DISTANCE | Distance between airports (miles) | Segment distance |

## Acquisition Method

The verification sample was obtained from the U.S. DOT Bureau of Transportation Statistics (BTS) TranStats download interface using the T-100 International Segment (All Carriers) table.

Source reference:

https://transtats.bts.gov/DL_SelectFields.aspx?QO_fu146_anzr=Nv4Pn44vr45&gnoyr_VQ=FJE

Download selections:

- Geography: unrestricted
- Year: 2024
- Period: January
- Fields: all available fields
- Carrier scope: all carriers
- Download format received: ZIP archive containing one CSV file
- Acquisition date: 2026-09-07

The original BTS archive is preserved unchanged in:

`data/raw/T_T100I_SEGMENT_ALL_CARRIER_20260907_133912.zip`

The archive contains:

`T_T100I_SEGMENT_ALL_CARRIER.csv`

Raw files under `data/raw/` are excluded from Git tracking.

The CSV is loaded directly from the ZIP archive with Pandas, so the raw BTS archive does not need to be extracted or modified.